src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1466,15 +1466,137 @@ public static void Sort(Array keys, Array? items, int index, int length, ICompar
 
             if (length > 1)
             {
-                SortImpl(keys, items, index, length, comparer ?? Comparer.Default);
+                comparer ??= Comparer.Default;
+
+                if (comparer == Comparer.Default)
+                {
+                    switch (Type.GetTypeCode(keys.GetType().GetElementType()))
+                    {
+                        case TypeCode.Boolean:
+                            if (TryGenericSort(keys as bool[], items, index, length)) return;
+                            break;
+                        case TypeCode.Byte:
+                            if (TryGenericSort(keys as byte[], items, index, length)) return;
+                            break;
+                        case TypeCode.Char:
+                            if (TryGenericSort(keys as char[], items, index, length)) return;
+                            break;
+                        case TypeCode.Double:
+                            if (TryGenericSort(keys as double[], items, index, length)) return;
+                            break;
+                        case TypeCode.Int16:
+                            if (TryGenericSort(keys as short[], items, index, length)) return;
+                            break;
+                        case TypeCode.Int32:
+                            if (TryGenericSort(keys as int[], items, index, length)) return;
+                            break;
+                        case TypeCode.Int64:
+                            if (TryGenericSort(keys as long[], items, index, length)) return;
+                            break;
+                        case TypeCode.SByte:
+                            if (TryGenericSort(keys as sbyte[], items, index, length)) return;
+                            break;
+                        case TypeCode.Single:
+                            if (TryGenericSort(keys as float[], items, index, length)) return;
+                            break;
+                        case TypeCode.String:
```

Comment:
Was it intentional to include String here?

I believe that there are subtle differences in how the current culture is handled between the non-generic and generic default comparer, so using this path for string may change the behavior.

Reply:
I'll just remove this case then. Thanks.